modules/tinymce/src/core/test/ts/browser/content/InsertContentTest.ts

```diff
@@ -351,4 +351,83 @@ describe('browser.tinymce.core.content.InsertContentTest', () => {
     InsertContent.insertAtCaret(editor, ' <strong> a </strong> ');
     TinyAssertions.assertContent(editor, '<pre> <strong> a </strong> </pre>');
   });
+
+  it('TINY-6263: insertAtCaret - merge font-size spans', () => {
```

Comment:
Can you please add a test for this case, as given your changes I believe it may add a regression...
```
<strong>
  <span style="background-color: red;">
    <strong>test</strong>
  </span>
</strong>
```

The output should be this:
```
<strong>
  <span style="background-color: red;">test</span>
</strong>
```

- [ ] Add a test similar to the above to ensure this doesn't introduce a regression

Reply:
Actually nvm, I misread what it was doing sorry...